node/p2p/misc.go

```diff
@@ -50,8 +50,17 @@ func ConnectionManager(cfg Config) func(params.Bootstrappers) (coreconnmgr.ConnM
 }
 
 // ConnectionGater constructs a ConnectionGater.
-func ConnectionGater(ds datastore.Batching) (coreconnmgr.ConnectionGater, error) {
-	return conngater.NewBasicConnectionGater(ds)
+func ConnectionGater(cfg Config) func(datastore.Batching) (*conngater.BasicConnectionGater, error) {
+	return func(ds datastore.Batching) (*conngater.BasicConnectionGater, error) {
+		m, err := conngater.NewBasicConnectionGater(ds)
+		if err != nil {
+			return nil, err
+		}
+		for _, peer := range cfg.BlacklistPeers {
+			_ = m.BlockPeer(peer)
```

Comment:
The ConnGater takes a Datastore and all the blacklisted peers are getting persisted inside it. This means that each time you run a node, the peers from the config are getting stored, again and again.

Reply:
So if we store them in Datastore - there is no need to store them in the Config. 
The persisted blocklist can be accessed in the runtime, via accessing the ConnGater directly. No need for it to be a node build time option.

Reply:
So basically, this PR should only ensure the ConnGater is accessible from the node and that's it

Reply:
So, your suggestion is to only keep changes with moving from interface type to direct implementation type, right?

Reply:
Right

Reply:
Then I suggest to close this Pr in favour of another with suggested changes.